crates/sparrow-compiler/src/ast_to_dfg.rs

```diff
@@ -95,6 +95,73 @@ pub(super) fn ast_to_dfg(
     )
 }
 
+pub fn add_udf_to_dfg(
```

Comment:
This code is duplicated from the `function` call, but a lot of pieces from that are unnecessary given that this is exclusively called from python client, meaning args are already flattened and locations not applicable. So, it can probably be refactored to share some methods but not worth the redirection imo.